lkql/extensions/src/lkql-node_data.adb

```diff
@@ -98,6 +109,48 @@ package body LKQL.Node_Data is
                                   To_Text (Exception_Message (Error))));
    end Access_Node_Field;
 
+   -----------------------------
+   -- Check_And_Remove_Prefix --
+   -----------------------------
+
+   function Check_And_Remove_Prefix
+     (Ctx        : Eval_Context;
+      Name_Node  : L.Identifier;
+      Prefix     : Text_Type;
+      White_List : String_Set) return Text_Type
```

Comment:
Nit: It's a bit confusing that `String_Set` is a hash map of `Unbounded_Text` elements, not `String`

Reply:
Ok, I'm reusing the `String_Set` that Geoffrey declared. I'm inbetween, because `Text_Type` is in fact our string type for all intent and purposes, and is really a string type (alias for Unbounded_Wide_Wide). I find `Text_Set` to be a bit less clear, but I guess I wouldn't mind much. Go ahead and rename it if you think it's worth it